.github/workflows/reusable-build-docs.yaml

```diff
@@ -0,0 +1,120 @@
+---
+name: Build Ansible package docs
+
+"on":
+  workflow_call:
+    inputs:
+      repository-owner:
+        description: GitHub account or org that owns the repository
+        type: string
+        required: false
+        default: ansible
+      repository-name:
+        description: Name of the GitHub repository
+        type: string
+        required: false
+        default: ansible-documentation
+      repository-branch:
+        description: Branch, tag, or commit SHA
+        type: string
+        required: false
+        default: devel
+      ansible-package-version:
+        type: string
+        description: Ansible community package version
+        required: false
+        default: devel
+    secrets:
+      DOCS_BOT_TOKEN:
+        required: true
+
+env:
+  PACKAGE_VERSION: ${{ inputs.ansible-package-version }}
+
+jobs:
+  build-package-docs:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Ansible documentation
+      uses: actions/checkout@v4
+      with:
+        repository: >-
+          ${{
+            inputs.repository-owner
+          }}/${{
+            inputs.repository-name
+          }}
+        ref: ${{ inputs.repository-branch }}
+        path: build-directory
+
+    - name: Setup nox
+      uses: wntrblm/nox@2024.10.09
+
+    - name: Output Python info
+      run: python --version --version && which python
+
+    - name: Graft ansible-core
+      run: nox -s clone-core
+      working-directory: build-directory
+
+    - name: Install project requirements
+      run: >-
+        python -m pip install
+        -r tests/requirements.in
+        -c tests/requirements.txt
+      working-directory: build-directory
+
+    - name: Set the COLLECTION_LIST variable
+      if: env.PACKAGE_VERSION != 'devel'
+      run: >-
+        echo COLLECTION_LIST="${PACKAGE_VERSION}"
```

Comment:
I'm not sure why this was originally named `COLLECTION_LIST`, since this has nothing to do with a list of collections, but is simply the package version. How about changing `COLLECTION_LIST` to `ANSIBLE_VERSION`?

Reply:
Hmm. The reason for the `COLLECTION_LIST` naming is probably lost to history. Maybe @samccann knows but I've accepted your other suggestion anyway so we can get rid of this.

Reply:
So... I've been banging away with that suggestion @felixfontein but it keeps failing with this error:

/home/runner/work/_temp/59fbd144-c21d-48bb-bfa4-6ee95c15c297.sh: line 1: ${env.PACKAGE_VERSION != 'devel' && env.PACKAGE_VERSION || '' }: bad substitution

Maybe we should just change to `ANSIBLE_VERSION`

Reply:
ah, OK. got it. I think it was a syntax issue. this seems to work:

```
ANSIBLE_VERSION="${{ env.PACKAGE_VERSION != 'devel' && env.PACKAGE_VERSION || '' }}"
```

Reply:
Ah sorry, I forgot to adjust the number of braces :facepalm: I'm glad you managed to figure this out :)